docs/madr/decisions/005-policy-matching.md

```diff
@@ -0,0 +1,1052 @@
+
+```yaml
+to:
+  - targetRef:
+      kind: MeshHTTPRoute
+      name: payments-get-info-v1-httproute
+    conf: ... # conf for "/v1/getinfo/" route of "payments" outbound 
+```
+
+When several targetRefs select the same outbound,
+corresponding confs are merged in the order they're presented in the "to" array 
+(bottom ones have more priority over the top ones):
+
+"backend" outbound is selected by 3 targetRefs
+
+```yaml
+to:
+  - targetRef: 
+      kind: Mesh
+      name: mesh-1
+    conf:
+      param1: value1
+  - targetRef: 
+      kind: MeshService
+      name: backend
+    conf:
+      param1: value2
+      param2: value3
+  - targetRef: 
+      kind: Mesh
+      name: mesh-1
+    conf:
+      param2: value4
+```
+
+resulting configuration for "backend" outbound is:
+
+```yaml
+conf:
+  param1: value2 # overrides 'value1' from 'targetRef{kind:Mesh}' 
+  param2: value4 # overrides 'value3' from 'targetRef{kind:MeshService,name:backend}'
+```
+
+#### Inbound policy
+
+All inbound policies contains a "from" array.
+Each element of the array has a "targetRef" that selects source of the traffic
+and has a configuration for selected source:
+
+```yaml
+from:
+  - targetRef: Mesh | MeshSubset | MeshService | MeshServiceSubset
+    conf: ...
+```
+
+Policy can select traffic from all sources:
+
+```yaml
+from:
+  - targetRef:
+      kind: Mesh
+      name: mesh-1
+    conf: ... # conf for all incoming traffic
+```
+
+Policy can select traffic from proxies with "kuma.io/zone: us-east" tag:
+
+```yaml
+from:
+  - targetRef:
+      kind: MeshSubset
+      name: mesh-1
+      tags:
+        kuma.io/zone: us-east
+    conf: ... # conf for traffic coming from "us-east" zone
+```
+
+Policy can select traffic from "backend" service:
+
+```yaml
+from:
+  - targetRef:
+      kind: MeshService
+      name: backend
+    conf: ... # conf for traffic coming from "backend" service
+```
+
+Policy can select traffic from instances of "backend" service that have version=v2:
+
+```yaml
+from:
+  - targetRef:
+      kind: MeshServiceSubset
+      name: backend
+      tags:
+        version: v2
+    conf: ... # conf for traffic coming from "backend" v2 service
+```
+
+Just like with outbound policies, when several targetRefs select the same source,
+corresponding confs are merged in the order they're presented in the "from" array
+(bottom ones have more priority over the top ones):
+
+Traffic from "backend" of v2 is selected by 3 targetRefs
+
+```yaml
+from:
+  - targetRef: 
+      kind: MeshServiceSubset
+      name: backend
+      tags:
+        version: v2
+    conf:
+      param1: value1
+  - targetRef:
+      kind: MeshService
+      name: backend
+    conf:
+      param1: value2
+      param2: value3
+  - targetRef:
+      kind: Mesh
+      name: mesh-1
+    conf:
+      param2: value4
+```
+
+resulting configuration for "backend" outbound is:
+
+```yaml
+conf:
+  param1: value2 # overrides 'value1' from 'targetRef{kind:Mesh}'
```

Comment:
value1 is in `targetRef: MeshServiceSubset` not `targetRef{kind:Mesh}` right?

Reply:
Good catch, fixed!